scripts/hal_alignment/maf_to_fasta.py

```diff
@@ -0,0 +1,246 @@
+            idx_dir_path = out_dir_path / map_uint_to_path(idx)
+            os.makedirs(idx_dir_path, exist_ok=True)
+
+            ga_recs = []
+            fasta_file_path = idx_dir_path / f'{idx}.fa'
+            with open(fasta_file_path, 'w') as out_f:
+                for rec in msa:
+                    maf_src = rec.id
+
+                    try:
+                        genome_name, dnafrag_name = maf_src_map[maf_src]
+                    except KeyError:
+                        try:
+                            genome_name, dnafrag_name = maf_src.split('.')
+                            maf_src_map[maf_src] = (genome_name, dnafrag_name)
+                        except ValueError as exc:
+                            if genomes_file is None:
+                                raise ValueError(
+                                    "MAF src field parse failed due to multiple dot separators"
+                                    " - please set genome names with the '--genomes-file' parameter"
+                                ) from exc
+
+                            match = maf_src_regex.match(maf_src)
+                            try:
+                                genome_name = match['genome']  # type: ignore
+                                dnafrag_name = match['seqid']  # type: ignore
+                            except TypeError as exc:
+                                raise ValueError(
+                                    "MAF src regex failed to parse MAF src field: '{maf_src}'") from exc
+
+                            maf_src_map[maf_src] = (genome_name, dnafrag_name)
+
+                    maf_start = rec.annotations['start']
+                    maf_size = rec.annotations['size']
+                    maf_end = maf_start + maf_size
+                    dnafrag_strand = rec.annotations['strand']
+
+                    if dnafrag_strand == 1:
+                        dnafrag_start = maf_start + 1
+                        dnafrag_end = maf_end
+                    else:
+                        maf_src_size = rec.annotations['srcSize']
+                        dnafrag_start = maf_src_size - maf_end + 1
+                        dnafrag_end = maf_src_size - maf_start
+
+                    fasta_id = f'{genome_name}:{dnafrag_name}:{dnafrag_start}:{dnafrag_end}'
+                    fasta_rec = SeqRecord(rec.seq, id=fasta_id, name='', description='')
+                    SeqIO.write([fasta_rec], out_f, 'fasta')
+
+                    ga_recs.append({
+                        'genome_name': genome_name,
+                        'dnafrag_name': dnafrag_name,
+                        'dnafrag_start': dnafrag_start,
+                        'dnafrag_end': dnafrag_end,
+                        'dnafrag_strand': dnafrag_strand
+                    })
+
+            json_file_path = idx_dir_path / f'{idx}.json'
+            with open(json_file_path, 'w') as f:
+                json.dump(ga_recs, f)
```

Comment:
```suggestion
                json.dump(ga_recs, f, indent=4)
```
Activates "pretty" dump format, so it is more human readable. If you like this change, I'm afraid you will need to reformat the test JSON files (sorry!).